src/server/game/Battlefield/Battlefield.cpp

```diff
@@ -267,26 +267,19 @@ void Battlefield::InvitePlayersInQueueToWar()
 {
     for (uint8 team = 0; team < PVP_TEAMS_COUNT; ++team)
     {
-        for (auto itr = _playerQueue[team].begin(); itr != _playerQueue[team].end();)
+        for (auto itr = _playerQueue[team].begin(); itr != _playerQueue[team].end(); itr = _playerQueue[team].erase(itr))
```

Comment:
do you *ALWAYS* erase? then don't erase at all, .clear()

Reply:
erase till the container is empty or cant invite cause battlefield is full

Reply:
maybe can rewrite it with deque and pop_front, what is more suitable?

Reply:
what is it currently?